csharp/src/Ice/Internal/ConnectionFactory.cs

```diff
@@ -1350,6 +1350,29 @@ public override void message(ThreadPoolCurrent current)
                 {
                     transceiver = _acceptor.accept();
 
+                    if (_maxConnections > 0 && _connections.Count == _maxConnections)
+                    {
+                        // Can't accept more connections, so we abort this transport connection.
+
+                        if (_instance.traceLevels().network >= 2)
+                        {
+                            _instance.initializationData().logger.trace(
+                                _instance.traceLevels().networkCat,
+                                $"rejecting new {_endpoint.protocol()} connection at {_acceptor} because the maximum number of connections is reached");
```

Comment:
```suggestion
                                $"rejecting new {_endpoint.protocol()} connection at {_acceptor} because the maximum number of connections has been reached");
```
The message as worded isn't grammatically correct, but since this is a log message, it doesn't really matter.
If you think a concise message is better than a correct message, I wouldn't blame you, and ignore my comment.